crates/walrus-service/src/node/blob_event_processor.rs

```diff
@@ -316,12 +391,58 @@ impl BlobEventProcessor {
             )
             .inc();
 
+            // Increment the counter and create a guard that will decrement it when dropped
+            let current_pending_event_count = self.background_per_processor_pending_event_count
```

Comment:
Nit: As we use `self.background_per_processor_pending_event_count[processor_index]` multiple times, it might be nicer to store it in a local variable.